scripts/pyenv_setup.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+# This script correctly sets up pyenv
+#
+# Assumptions:
+# - This script assumes you're calling from the top directory of the repository
+set -eu
+
+# Check if a command is available
+require() {
+    command -v "$1" >/dev/null 2>&1
+}
+
+query_big_sur() {
+    if require sw_vers && sw_vers -productVersion | grep -E "11\." > /dev/null; then
+        return 0
+    fi
+    return 1
+}
+
+get_shell_startup_script() {
+  local _startup_script=''
+  if [ -n "$SHELL" ]; then
```

Comment:
Since bash is the shebang you might as well use bash features, like `[[`. `[` isn't a builtin, it's an external program.

Reply:
[[ is better, but it's worth pointing out that [ is in fact a builtin in bash and most other Bourne-like shells, meaning that it doesn't call fork/exec. [[ is a keyword, meaning it has special abilities, such as allowing the argument to be unquoted (and in fact requiring it depending on your usage).

Reply:
Haha, I stand corrected, my cmistake was to use `which` which inspects PATH whereas the preferred `command -v` which I don't enjoy typing out cause it's longer will tell you more what the shell'l do. Thanks.

```
$ docker run --rm -it debian:10-slim
root@3c380936fbf7:/# echo $SHELL
/bin/bash
root@3c380936fbf7:/# which [
/usr/bin/[
root@3c380936fbf7:/# command -v [
[
root@3c380936fbf7:/# cat lol.sh
[ 1 = 1 ] && echo hi
root@3c380936fbf7:/# strace bash lol.sh 2>&1 | grep exec
execve("/bin/bash", ["bash", "lol.sh"], 0x7ffe89875c58 /* 7 vars */) = 0
```

Reply:
I still stand by `[[` as an indicator you're using bash, and not a mystery sh.